### PR TITLE
Adding guard and validators

### PR DESCRIPTION
**Pull request by Rominos7**
**Jira ticket**
 NONE
 
 **Description**
 - Work with comments
 - Added validators for validation request data(endpoints and DB)

### CODE REVIEW

File: src/tasks/tasks.service.ts [hunk 1]
```diff
@@ -24,37 +17,23 @@ export class TasksService {
...
-        console.log('Task was updated', taskToUpdate);
-
-        return this.tasksRepostiory.save(taskToUpdate);
-    }
-    throw new HttpException('ID not found', HttpStatus.NOT_FOUND);
-    
+  async updateTask(id:ValidateId) {
+    let taskToUpdate: RequestIsNotEmpty = await this.tasksRepostiory.findOne(id);
+    // TODO: For more changes to update need to expand this "line"
+    taskToUpdate.status = TaskStatus.Solved;
```
Comment (Rominos7) [inline src/tasks/tasks.service.ts:31]: Comment was for line
**taskToUpdate.status = TaskStatus.Solved**

File: src/tasks/tasks.guard.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { Observable } from 'rxjs';
+
+// this was my work with guards. Nothing fansy. 
+@Injectable()
+export class TasksGuard implements CanActivate {
+    canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
```
Comment (Rominos7) [inline src/tasks/tasks.guard.ts:10]: This was my attempt to work with guards. So I created one and used it in controller.

File: src/tasks/tasks.validators.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+export class ValidateId {
+  @IsNotEmpty()
+  @IsNumberString()
+  id: number;
+}
+
+export class RequestIsNotEmpty {
+    @IsNotEmpty() id: number;
+    @IsNotEmpty() status: string;
```
Comment (Rominos7) [inline src/tasks/tasks.validators.ts:20]: Maybe we need to add some more "validation" to the fileds? Not just **IsNotEmpty** ?

Comment (Rominos7) [inline src/tasks/tasks.validators.ts:20]: I added validation for checking is this request is not empty (by checking every field). Is there more proficient way to do so?

Also does I need to do more validation checks when working with DB?

**Comment by pshata** [comment]
I see that you added validation class and it looks good. Our goal was to setup validation pipe. I created PR - [link](https://github.com/Rominos7/Node_nest/pull/2) with an example. Please check when you'll have a time. Other we can dicsuss in our sync.

**Pull request merged**